src/properties/mod.rs

```diff
@@ -351,10 +352,26 @@ pub struct PropertyOptions<'a> {
     pub hints: &'a HashMap<String, String>,
     /// Tracks the property tree location in a GVAS file.
     pub properties_stack: &'a mut Vec<String>,
+    /// Custom versions
+    pub custom_versions: &'a [FCustomVersion],
     /// Enables large world coordinates.
     pub large_world_coordinates: bool,
 }
 
+impl<'a> PropertyOptions<'a> {
+    /// Get custom version
+    pub fn get_custom_version<T>(&self) -> FCustomVersion
+    where
+        T: CustomVersionTrait + Into<i32>,
+    {
+        self.custom_versions
+            .iter()
+            .find(|e| e.key == T::GUID)
+            .cloned()
+            .unwrap_or_else(|| FCustomVersion::new(T::GUID, 0))
+    }
```

Comment:
I think this fun could return an `Option<&FCustomVersion>` to eliminate the clone and the unwrap. Consider replacing with something like this:
```suggestion
    pub fn get_custom_version<T>(&self) -> Option<&FCustomVersion>
    where
        T: CustomVersionTrait + Into<i32>,
    {
        self.custom_versions
            .iter()
            .find(|e| e.key == T::GUID)
    }
```

Reply:
It would be a bit more work on the consumer's side to check the version, I would like this API to be as straightforward to the user as possible, default initialization in this case seems acceptable